File: app/src/main/java/com/cheatdatabase/cheat_detail_view/MemberCheatViewFragmentAdapter.java
package com.cheatdatabase.cheat_detail_view;

import android.widget.LinearLayout;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.cheatdatabase.model.Cheat;

import java.util.List;

public class MemberCheatViewFragmentAdapter extends FragmentPagerAdapter {
    private final LinearLayout outerLayout;
    private List<Cheat> cheats;

    public MemberCheatViewFragmentAdapter(FragmentManager fragmentManager, List<Cheat> cheats, LinearLayout outerLayout) {
        super(fragmentManager);
        this.cheats = cheats;
        this.outerLayout = outerLayout;
    }

    @Override
    public Fragment getItem(int position) {
        return MemberCheatViewFragment.newInstance(cheats, position, outerLayout);
    }

    @Override
    public int getCount() {
        return cheats.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return cheats.get(position % cheats.size()).getCheatTitle();
    }
}

File: app/src/main/java/com/cheatdatabase/cheatdetailview/MemberCheatViewFragmentAdapter.kt
package com.cheatdatabase.cheatdetailview

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.cheatdatabase.cheatdetailview.MemberCheatViewFragment.Companion.newInstance
import com.cheatdatabase.data.model.Cheat

class MemberCheatViewFragmentAdapter(
    fragmentActivity: FragmentActivity, private val cheats: List<Cheat>
) : FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return newInstance(cheats, position)
    }

    override fun getItemCount(): Int {
        return cheats.size
    }

//    override fun getPageTitle(position: Int): CharSequence? {
//        return cheats[position % cheats.size].cheatTitle
//    }
}